tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/ElasticsearchReindexingTests.java

```diff
@@ -0,0 +1,263 @@
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(ElasticsearchReindexingTests.class);
+
+    public static final String CUSTOM_ALFRESCO_INDEX = "custom-alfresco-index";
+
+    @Autowired
+    private ServerHealth serverHealth;
+    @Autowired
+    private DataUser dataUser;
+    @Autowired
+    private DataSite dataSite;
+    @Autowired
+    private DataContent dataContent;
+    @Autowired
+    protected RestWrapper client;
+
+    private UserModel testUser;
+
+    private SiteModel testSite;
+
+    private RestHighLevelClient elasticClient;
+
+    /**
+     * Create a user and a private site and wait for these to be indexed.
+     */
+    @BeforeClass(alwaysRun = true)
+    public void dataPreparation()
+    {
+        serverHealth.isServerReachable();
+        serverHealth.assertServerIsOnline();
+
+        Step.STEP("Create a test user and private site containing a document.");
+
+        testUser = dataUser.createRandomTestUser();
+        testSite = dataSite.usingUser(testUser).createPrivateRandomSite();
+        createDocument();
+
+        Step.STEP("create ES client");
+        elasticClient = new RestHighLevelClient(
+                RestClient.builder(new HttpHost(AlfrescoStackInitializer.elasticsearch.getContainerIpAddress(),
+                                                AlfrescoStackInitializer.elasticsearch.getFirstMappedPort(),
+                                                "http")));
+
+    }
+
+    /**
+     * This is run as the first test in the class so that we know that no other test has indexed the system documents.
+     */
+    @Test(groups = TestGroup.SEARCH, priority = -1)
+    public void testReindexerIndexesSystemDocuments() throws Exception
+    {
+        LOGGER.info("Starting test");
+        // GIVEN
+        // Check a particular system document is NOT indexed.
+        // Nb. The cm:name:* term ensures that the query hits the index rather than the db.
+        String queryString = "=cm:name:budget.xls AND =cm:title:\"Web Site Design - Budget\" AND cm:name:*";
+        expectResultsFromQuery(queryString, dataUser.getAdminUser());
+
+        // WHEN
+        // Run reindexer against the initial documents.
+        reindex(Map.of("ALFRESCO_REINDEX_JOB_NAME", "reindexByIds",
+                       "ELASTICSEARCH_INDEX_NAME", CUSTOM_ALFRESCO_INDEX,
+                       "ALFRESCO_REINDEX_FROM_ID", "0",
+                       "ALFRESCO_REINDEX_TO_ID", "1000"));
+
+        // THEN
+        // Check system document is indexed.
+        Utility.sleep(1000, 10000, () -> {
+            expectResultsFromQuery(queryString, dataUser.getAdminUser(), "budget.xls");
+        });
+        LOGGER.info("End test");
+    }
+
+    @Test(groups = TestGroup.SEARCH)
+    public void testReindexerFixesBrokenIndex() throws Exception
+    {
+        // GIVEN
+
+        // Delete all documents inside Elasticsearch.
+        cleanUpIndex();
```

Comment:
Why do we need to delete all documents from the index here?

Reply:
to be sure we restart from scratch